### PR TITLE
Model test and implementation

### PR DESCRIPTION
**Pull request by rrustia**

### CODE REVIEW

File: server/model.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
     // progress: { 
-    //        phildor : completed,
-    //        alekhein: not completed
+    //        phildor : false, 
+    //        alekhein: true
     // }
-    progress: { type: Object },
-    index: encrypt(email) // use bcryptjs or create own, will be used for indexing and access purpose
+    email: { type: String, required: true },
+    progress: { opening: { type: String }, complete: { type: Boolean} }
```
Comment (rrustia) [inline server/model.js]: Changed "progress" object structure.

File: server/.nyc_output/24dafba436ba89efff208ca93c492d79.json [hunk 2]
```diff
@@ -0,0 +1 @@
...7},"end":{"line":1,"column":36}},"1":{"start":{"line":2,"column":0},"end":{"line":2,"column":34}},"2":{"start":{"line":3,"column":19},"end":{"line":13,"column":2}},"3":{"start":{"line":15,"column":0},"end":{"line":19,"column":2}},"4":{"start":{"line":16,"column":4},"end":{"line":18,"column":6}},"5":{"start":{"line":21,"column":0},"end":{"line":25,"column":2}},"6":{"start":{"line":22,"column":4},"end":{"line":24,"column":6}},"7":{"start":{"line":27,"column":13},"end":{"line":27,"column":47}},"8":{"start":{"line":29,"column":0},"end":{"line":29,"column":24}}},"fnMap":{"0":{"name":"(anonymous_0)","decl":{"start":{"line":15,"column":29},"end":{"line":15,"column":30}},"loc":{"start":{"line":15,"column":40},"end":{"line":19,"column":1}},"line":15},"1":{"name":"(anonymous_1)","decl":{"start":{"line":21,"column":34},"end":{"line":21,"column":35}},"loc":{"start":{"line":21,"column":45},"end":{"line":25,"column":1}},"line":21}},"branchMap":{},"s":{"0":1,"1":1,"2":1,"3":1,"4":1,"5":1,"6":1,"7":1,"8":1},"f":{"0":1,"1":1},"b":{},"_coverageSchema":"332fd63041d2c1bcb487cc26dd0d5f7d97098a6c","hash":"4df334b76e5b4495eb07002c0c78ca3fbe8595ef","contentHash":"1fb08f6c1d39fdef948bf8e55c01e30c_11.4.1"}}
```
Comment (visaudi) [inline server/.nyc_output/24dafba436ba89efff208ca93c492d79.json]: Take out NYC output.

File: server/model.js [hunk 3]
```diff
@@ -1,26 +1,29 @@
-const bcrypt = require('bcryptjs');
 const mongoose = require('mongoose');
-
 mongoose.Promise = global.Promise;
-
 const UserSchema = mongoose.Schema({
     
     // "progress" loaded automatically on successful login
```
Comment (visaudi) [inline server/model.js]: Indent should be four or two spaces. Preferably four.

File: server/model.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    it("should be invalid if email is empty", () => {
+        chai.assert.isTrue(testUser.email == testEmail);
+    });
+
+    it("apiRepr method should return email string", () => {
+        chai.assert.isTrue(JSON.stringify(testUser.apiRepr()) == JSON.stringify({ email: testEmail }), "is not true" );
+    })
+
+    it("progressRepr method should return an object with opening data", () => {
+        chai.assert.isTrue(JSON.stringify(testUser.progressRepr().progress) == JSON.stringify({ opening: testOpening, complete: testBool }), "is not true" );
```
Comment (visaudi) [inline server/model.test.js]: Indent.

Comment (visaudi) [inline server/model.test.js]: Store each stringified object to a variable.

File: server/model.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        const expected = JSON.stringify(testUser.apiRepr());
+        const testResult = JSON.stringify({ email: testEmail });
+        chai.assert.isTrue(expected == testResult, "is not true" );
+    })
+
+    it("progressRepr method should return an object with opening data", () => {
+        const expected = JSON.stringify(testUser.progressRepr().progress);
+        const testResult = JSON.stringify({ 
+            opening: testOpening, 
+            complete: testBool });
```
Comment (visaudi) [inline server/model.test.js]: Can you make the curly brace and paren suggest the current scope using the next line?

**Pull request merged**